docs/learn/WSL及docker的安装 .md

```diff
@@ -0,0 +1,188 @@
+
+`dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart`
+
+
+
+此步可能报错：<font style="color:rgb(79, 79, 79);">无法解析服务器的名称或地址</font>
+
+<font style="color:rgb(77, 77, 77);">解决办法是修改域名DNS</font>
+
+按`<font style="background-color:#FFFFFF;">windows + i</font>`快捷键打开设置，在左侧菜单栏找到网络和internet
+
+<font style="color:rgba(0, 0, 0, 0.75);">点击高级网络设置</font>
+
+<font style="color:rgba(0, 0, 0, 0.75);">选中当前使用的网络，点击查看其它属性，之后点击编辑</font>
+
+<font style="color:rgba(0, 0, 0, 0.75);">修改dns，将自动代理改为手动IPv4，输入114.114.114.114</font>
+
+<font style="color:rgba(0, 0, 0, 0.75);"></font>
+
+**<font style="color:rgba(0, 0, 0, 0.75);">启用虚拟机平台</font>**
+
+<font style="color:rgb(25, 27, 31);">在 Windows 10（2004）上启用虚拟机平台</font>
+
+`dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart`
+
+<font style="color:rgb(25, 27, 31);">在 Windows 10（1903，1909）上启用虚拟机平台</font>
+
+`Enable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform -NoRestart`
+
+
+
+可能报错：0x800701bc
+
+终端输入wsl --update即可，更新一下wsl
+
+可能报错：wsl更新进度为一直为0或者不变
+
+`wsl --install --web-download`
+
+让他从github下载，或者重启一下试试
+
+
+
+**<font style="color:rgb(25, 27, 31);">设置 WSL 2 为默认值</font>**
+
+`wsl --set-default-version 2`
+
+
+
+**<font style="color:rgb(25, 27, 31);">安装一个 Linux 发行版</font>**
+
+<font style="color:rgb(25, 27, 31);">微软应用商店搜索ubuntu20.04下载即可</font>
+
+<font style="color:rgb(25, 27, 31);">下载好后打开设置用户名密码，WSL的安装到这里就结束了</font>
+
+<font style="color:rgb(25, 27, 31);"></font>
+
+### <font style="color:rgb(25, 27, 31);">三.安装docker及安装时可能有的报错</font>
+建议装的时候全程开代理TUN模式
+
+**首先卸载可能会冲突的软件包**
+
+打开下载的ubuntu后输入
+
+`for pkg in docker.io docker-doc docker-compose docker-compose-v2 podman-docker containerd runc;do sudo apt-get remove $pkg;done`
+
+
+
+**使用存储库安装apt**
+
+```powershell
+# Add Docker's official GPG key
+sudo apt-get update
+sudo apt-get install -y ca-certificates curl
+sudo install -m 0755 -d /etc/apt/keyrings
+sudo curl -fsSL https://mirrors.openctf.dev/https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://mirrors.openctf.dev/https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt-get update
+```
+
+
+
+**安装 Docker 软件包**
+
+`sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin`
+
+
+
+可能的报错：_**GPG error**_
+
+下载gpg
+
+```powershell
+sudo install -m 0755 -d /etc/apt/keyrings
+sudo curl -fsSL https://mirrors.openctf.dev/https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+sudo chmod a+r /etc/apt/keyrings/docker.asc
+```
+
+[GPG（GnuPG）的安装和使用_gpg软件安装-CSDN博客](https://blog.csdn.net/huangjc0715/article/details/114778912)
+
+下载好后回到再从安装apt这一步开始即可
```

Comment:
gpg installation is listed above the former step. this error is because you didn't successfully curl the gpg key at the former step.